opencti-platform/opencti-graphql/src/schema/schema-relationsRef.ts

```diff
@@ -1,104 +1,118 @@
+  relationsRefCacheArray: new Map<string, RelationRefDefinition[]>(),
+  relationsRefCacheMap: new Map<string, Map<string, RelationRefDefinition>>(),
 
   registerRelationsRef(entityType: string, relationsRefDefinition: RelationRefDefinition[]) {
-    this.relationsRef[entityType] = [...this.relationsRef[entityType] ?? [], ...relationsRefDefinition];
+    const directRefs = this.relationsRef[entityType] ?? new Map<string, RelationRefDefinition>();
 
+    // Register given relations ref
     relationsRefDefinition.forEach((relationRefDefinition) => {
-      if (!this.inputNames.includes(relationRefDefinition.inputName)) {
-        this.inputNames.push(relationRefDefinition.inputName);
-        this.databaseNames.push(relationRefDefinition.databaseName);
-        this.stixNames.push(relationRefDefinition.stixName);
-        if (relationRefDefinition.checker) {
-          this.registerChecker(relationRefDefinition.databaseName, relationRefDefinition.checker);
-        }
-        if (relationRefDefinition.multiple) {
-          this.databaseNameMultiple.set(relationRefDefinition.databaseName);
-          this.inputNameMultiple.set(relationRefDefinition.inputName);
-        }
-        this.databaseNameToInputName[relationRefDefinition.databaseName] = relationRefDefinition.inputName;
-        this.inputNameToDatabaseName[relationRefDefinition.inputName] = relationRefDefinition.databaseName;
-        this.stixNameToInputName[relationRefDefinition.stixName] = relationRefDefinition.inputName;
-        this.inputNameToStixName[relationRefDefinition.inputName] = relationRefDefinition.stixName;
-        this.relationsRef_by_input[relationRefDefinition.inputName] = relationRefDefinition;
+      // Check name collision with STIX_CORE_RELATIONSHIP
+      if (STIX_CORE_RELATIONSHIPS.includes(relationRefDefinition.databaseName)) {
+        throw UnsupportedError('You can\'t register a relations ref with an existing stix-core-relationship name', {
+          relationRef: relationRefDefinition.databaseName
+        });
+      }
+
+      // Check duplicate relations ref
+      if (directRefs.has(relationRefDefinition.inputName)) {
+        throw UnsupportedError('You can\'t register two relations ref with the same name on an entity', {
+          relationRef: relationRefDefinition.inputName,
+          entityType
+        });
       }
+
+      directRefs.set(relationRefDefinition.inputName, relationRefDefinition);
     });
+
+    this.relationsRef[entityType] = directRefs;
+
+    this.computeCache(entityType);
   },
 
-  getRelationsRefByInputName(inputName: string): RelationRefDefinition {
-    return this.relationsRef_by_input[inputName];
+  // Extract this method to be call in all methods
+  // When an entity not register any relations, the relations for this entity is not computed
+  // Call only in register mechanism when all the entities will be migrated
+  computeCache(entityType: string) {
```

Comment:
Same remark than the other files.
By the way why not just create all the map at registration?